compiler-rt/lib/builtins/extendhfxf2.c

```diff
@@ -0,0 +1,16 @@
+//===-- lib/extendhfxf2.c - half -> long double conversion --------*- C -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include "int_lib.h"
+#define SRC_HALF
+#define DST_DOUBLE
+#include "fp_extend_impl.inc"
+
+// Long double are expected to be as precise as double.
```

Comment:
Does this need to be guarded with `#if HAS_80_BIT_LONG_DOUBLE && defined(COMPILER_RT_HAS_FLOAT16)`? I'm not sure if i386 supports float16.

Reply:
That gets handled by https://github.com/llvm/llvm-project/blob/50e73aeea2a7e2cbaeb66252aadea9a6144937df/compiler-rt/lib/builtins/fp_extend.h#L56C8-L56C31 where the src_t would be uint16_t if Float16 is not available. Even extendhfsf2 does not have this check.

Reply:
Sounds good to me